File: client/JSON/Reciept.ts
import { Tax } from "../Types/Tax";

/**
 * Represents the reciept
 * 
 * @author Melkvi Dániel - h879476
 * @version 0.1
 * @since 0.1
 */
export interface Reciept{
    Entries: RecieptEntry[];
    Sum: number;
    Tax: Tax;
}

/**
 * Represents an entry in the reciept
 * 
 * @author Melkvi Dániel - h879476
 * @version 0.1
 * @since 0.1
 */
export interface RecieptEntry{
    Name: string;
    Count: number;
    Value: number;
}

File: server/DAL/JSON/Reciept.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Etel.DAL.JSON
{
    [Serializable]
    public class Reciept
    {
        public List<RecieptEntry> Entries { get; set; }
        public int Sum { get; set; }

        public Tax Tax { get; set; }
    }

    [Serializable]
    public class RecieptEntry
    {
        public string Name { get; set; }
        public int Count { get; set; }
        public int Value { get; set; }
    }
}
